docs/hub/how-to-downstream.md

```diff
@@ -0,0 +1,53 @@
+# How to integrate downstream methods in your library
+
+In the tutorial, you learned some basic functions for integrating the Hub into your library. Functions that allow your library to download files from the Hub are referred to as *downstream* functions. This guide introduces additional downstream methods you can integrate in your library. You will learn how to:
+
+* Download a file without caching it on your disk.
+* Download a file and cache it on your disk.
+* Download all the files in a repository.
+
+## `hf_hub_url`
+
+Use `hf_hub_url` to retrieve the URL of a specific file to download by providing a `filename`.
+
+![/docs/assets/hub/repo.png](/docs/assets/hub/repo.png)
+
+```python
+>>> from huggingface_hub import hf_hub_url
+>>> hf_hub_url(repo_id="lysandre/arxiv-nlp", filename="config.json")
+'https://huggingface.co/lysandre/arxiv-nlp/resolve/main/config.json'
+```
+
+Specify a particular file version by providing the file revision. The file revision can be a branch, a tag, or a commit hash:
```

Comment:
What do you think of mentioning that the commit hash has to be full-length and not the usual 6-7 characters short SHA? We've had issues in the past were users were surprised that the short SHA didn't work

Reply:
Definitely worth mentioning to prevent unwanted surprises :)